### ISSUE TITLE
Preload Steam Overlay

### ISSUE THREAD

**Issue by Lucki** [lhark/truckersmp-cli#39]
Since we're using steam anyway we can also use it's overlay.

~~~ patch
diff --git a/truckersmp-cli b/truckersmp-cli
index 059995d..f518f59 100755
--- a/truckersmp-cli
+++ b/truckersmp-cli
@@ -149,6 +149,7 @@ start_with_proton() {
 
     SteamGameId="$steamid" \
     SteamAppId="$steamid" \
+    LD_PRELOAD="${LD_PRELOAD}${LD_PRELOAD:+:}$steamdir/ubuntu12_64/gameoverlayrenderer.so" \
     STEAM_COMPAT_DATA_PATH="$prefixdir" \
     STEAM_COMPAT_CLIENT_INSTALL_PATH="$steamdir" \
     /usr/bin/env python3    "$protondir/proton" \
~~~

**Comment by TheGreatestJannet** [comment]
Has this been merged into the main code? I'm asking this as currently Steam Overlay doesn't work for me. If not do you have any plans to do so? Thank you.

**Comment by tianyu313588** [comment]
Can you tell me how to use it? I read the source code for a day.

**Comment by Lucki** [comment]
~~~
$ patch -i difffile.patch
patching file truckersmp-cli
Hunk #1 succeeded at 148 with fuzz 2 (offset -1 lines).
~~~

**Comment by tianyu313588** [comment]
Do I need to modify the source code or write another C file?

**Comment by Lucki** [comment]
Just create a `difffile.patch` file with the content above.

**Comment by Lucki** [comment]
Implemented in #31.
